### PR TITLE
fw-4200, support multiple categories for dict entries

### PR DESCRIPTION
**Pull request by sarahfirstvoices**
### Description of Changes
* Update DictionaryEntry to support multiple categories
* Expand category data in the dictionary response data
* Adjusted related names for dictionary models to be clear which are to the m2m related model and which are to the through model
* Added some inline admin for related dictionary models
* Bonus: use a generic id-title-url serializer for the category response data. Updated urlconf basenames to work with this in future. (It is the recommended naming method anyway-- the model's object name.)

### Checklist
- [x] README / documentation has been updated if needed
- [x] PR title / commit messages contain Jira ticket number if applicable
- [x] Tests have been updated / created if applicable
- [x] Admin Panel has been updated if models have been added or modified
- [x] Migrations have been updated and committed if applicable

### Reviewers Should Do The Following:
- [x] Review the code changes here
- [ ] Pull the branch and test locally

### Additional Notes
N/A

### CODE REVIEW

File: firstvoices/backend/admin/dictionary_admin.py
Comment (bronson-c) [inline firstvoices/backend/admin/dictionary_admin.py:79]: +1 for adding these inlines

**Review by bronson-c — APPROVED** [review]
Great addition of the id-title-url serializer

File: firstvoices/backend/admin/dictionary_admin.py
Comment (NolanVH) [inline firstvoices/backend/admin/dictionary_admin.py:108]: I am getting a no `admin_link` error on dictionary entry admin pages because there is no reverse URL for the related characters admin page.

I think adding the following here should fix it:
`admin.site.register(DictionaryEntryRelatedCharacter, HiddenBaseAdmin)`

File: firstvoices/backend/models/dictionary.py
Comment (NolanVH) [inline firstvoices/backend/models/dictionary.py:198]: This could possibly use a string function here to improve readability, mainly in the admin panel, which returns something like `self.to_dictionary_entry.title`.

**Review by NolanVH — APPROVED** [review]
These changes look good to me, aside from the few minor issues in the comments! The few issues I found are mainly to do with the admin panel and could be changed without a second review if needed, so I will approve this PR.

Tested locally working as intended and tests are passing.
Thanks, @sarahfirstvoices!

File: firstvoices/backend/admin/dictionary_admin.py
Comment (sarahfirstvoices) [reply firstvoices/backend/admin/dictionary_admin.py:108]: Good catch, Nolan!

**Pull request merged**
